pkg/prometheus/promcfg_test.go

```diff
@@ -2704,3 +2704,35 @@ func TestHonorTimestamps(t *testing.T) {
 		}
 	}
 }
+
+func TestGetSampleLimit(t *testing.T) {
+	tcs := []struct {
+		Enforced uint64
+		Expected uint64
+		User     uint64
+	}{
+		{
+			Enforced: 100,
+			User:     1000,
+			Expected: 100,
+		},
+		{
+			Enforced: 0,
+			User:     888,
+			Expected: 0,
```

Comment:
That's not what I would have expected. `0` enforced means nothing is enforced, no? In that case if a user chooses a stricter limit, I would expect that limit to be used.

Reply:
Yes! Good catch!